### ISSUE TITLE
I think I find a little BUG for informations on moment with Templater

### ISSUE THREAD

**Issue by MugenLee** [SilentVoid13/Templater#372]
First, em, sorry my English is not very good, So, please just take a look...

The Bug is that:

when I use "tp.date.weekday" to change my "WeeklyNote", I find it can not support the "Week of Year". whether"WW","Wo","W","ww","wo",or "w"

like this imge: I want to position in "47W", but no matter how I enter it, it is always positioned in the first week of November
![![image](https://user-images.githubusercontent.com/91043731/134110453-518dc47a-160a-446c-bb12-26c517f8a81f.png)](https://user-images.githubusercontent.com/91043731/134110433-0bf4a8ee-f01e-4b59-b1fb-c30bde4ce2ea.png)

But when I add the "day" with moment "DD",it can position in "47W"
![image](https://user-images.githubusercontent.com/91043731/134110488-54667007-58bd-4728-9e4d-4ce2b4d3a75d.png)

I see it than I think it can support the "Week of Year"
![image](https://user-images.githubusercontent.com/91043731/134110507-1bc5a06c-6f1a-4a22-9cc8-f6fa01f24d67.png)

So, I think it is a little bug, please take a look it, thank you !

**Comment by MugenLee** [comment]
hi dear SilentVoid13, please take a look it. thank you.

This problem affected the reconstruction of my previous Weekly.
The previous weekly only recorded "WW", but Templater couldn't identify it , so the reconstruction is Stagnant.

You can contact me if you have any questions on it.

Looking forward to your reply.

**Comment by SilentVoid13** [comment]
Hey @MugenLee, this isn't a Templater bug, but a moment.js bug. Templater just calls moment.js under the hood, but that's it.

I made a lot of tests and it looks like the `MM` part is what makes moment.js go wrong.

```js
<% moment("47", "ww") %> // Correct
<% moment("47-2021", "ww-YYYY") %> // Correct
<% moment("47-25", "ww-DD") %> // Correct
<% moment("47-11", "ww-MM") %> // Incorrect
<% moment("47-11-2021", "ww-MM-YYYY") %> // Incorrect
<% moment("47-11-2021", "WW-mm-YYYY") %> // Correct
<% moment("47-11-2021-25", "ww-MM-YYYY-DD") %> // Correct, but we need to specify the day
```

One solution that is a bit ugly but that works is to purposely mistype the format for the month, for example replacing `MM` (month) with `mm` (minutes). This allows you to ignore the month part.
I tried searching for someone that already had this bug with moment, but couldn't find anything.

TLDR: You can make it work by replacing

```
<% tp.date.weekday("ww[W]YYYY-MM-DD", 0, tp.file.title, "ww-MM-YYYY") %>
```

with:

```
<% tp.date.weekday("ww[W]YYYY-MM-DD", 0, tp.file.title, "ww-mm-YYYY") %>
```

Closing this, since this isn't a templater bug. You can open an issue on the moment.js github if you wish.

**Comment by MugenLee** [comment]
wow, thank you for you answer this

After reading your explanation, I understand, it turns out that this is an
error of moment.js
Because it has nothing to do with TP, please close it
Let me study it again, the solution ideas and solutions you put forward are
very informative, thank you again

SilentVoid13 ***@***.***> 于2021年9月29日周三 下午6:31写道：
